examples/discovery-search-app/scripts/validateSdkEnvVars.js

```diff
@@ -0,0 +1,45 @@
+const fs = require('fs');
+const dotenv = require('dotenv');
+const path = require('path');
+const ibmCredentialsFilePath = path.join(__dirname, '../', 'ibm-credentials.env');
+
+function validateRequiredEnvVar(envVar) {
+  if (!process.env[envVar]) {
+    throw new Error(
+      `Please set ${envVar} in your environment or in ${ibmCredentialsFilePath} file`
+    );
+  }
+}
+
+function loadCredentialsFile() {
+  if (!process.env.IBM_CREDENTIALS_FILE && fs.existsSync(ibmCredentialsFilePath)) {
+    process.env.IBM_CREDENTIALS_FILE = ibmCredentialsFilePath;
+  }
+  if (process.env.IBM_CREDENTIALS_FILE) {
+    dotenv.config({
+      path: process.env.IBM_CREDENTIALS_FILE
+    });
+  }
+}
+
+function validateSdkEnvVars() {
+  try {
+    validateRequiredEnvVar('DISCOVERY_AUTH_TYPE');
+  } catch (e) {
+    loadCredentialsFile();
+    validateRequiredEnvVar('DISCOVERY_AUTH_TYPE');
+  }
+
+  if (process.env.DISCOVERY_AUTH_TYPE === 'cp4d') {
+    [
+      'DISCOVERY_URL',
```

Comment:
if we wanted to - we could remove this from the required env vars and just have the `setSdkUrl` take care of it if needed